docs/sequence-diagrams/compact.puml

```diff
@@ -0,0 +1,7 @@
+@startuml compact
+thanos_compact -> thanos_compact_config: Evaluate retention policy to see how long to keep samples
```

Comment:
This is not really what's happening in order. I think the easiest way for you would to be go to implementation and look it up. 

It is rather:
* Sync meta files
* Compact
   * plan
   * download planned blocks to be compacted
   * compact 
   * upload compacted
   * delete old
* Downsample
   * ... 
* Apply Retention 

I don't really get what `thanos_compact_ config` is - it's not really an actor for sequential graph.

This means that as this doc of steps is useful, not sure if sequential graph is useful. If any I would see 3 "actors". 
* Thanos compact process
* Object storage (let's name it like that)
* Local filesystem

What do you think?